tango_simlib/tests/test_fandango_json_parser.py

```diff
@@ -0,0 +1,183 @@
+                cmd_prop_value = cmd_metadata[cmd_prop]
+                self.assertEquals(
+                    str(cmd_prop_value).find('Const'), -1,
+                    "The value '%s' of the cmd property, '%s', for the command, '%s',"
+                    "has 'Const'" % (cmd_prop_value, cmd_prop, cmd_name)
+                    )
+                self.assertEquals(
+                    tango._tango.CmdArgType, type(cmd_prop_value),
+                    "The data type '%s' of the cmd property, '%s', of the command, '%s',"
+                    "is not a tango type" % (cmd_prop_value, cmd_prop, cmd_name)
+                    )
+
+    def test_preprocess_attribute_types(self):
+        """Testing values of attribute properties, 'data_type' and 'data_format',
+        of all attributes are changed to tango types.
+        """
+
+        # testing that data type of specified attr property values is a tango type
+        dev_attr = self.fandango_parser.get_device_attribute_metadata().items()
+        for attr_name, attr_metadata in dev_attr:
+            self.assertEquals(
+                tango._tango.CmdArgType,
+                type(attr_metadata['data_type']),
+                "The value of attr prop 'data_type' of attr name, '%s', is not a tango type."
+                % (attr_name)
+                )
+            self.assertEquals(
+                tango._tango.AttrDataFormat,
+                type(attr_metadata['data_format']),
+                "The value of attr prop, 'data_format', of attr name, '%s', is not a tango type."
+                % (attr_name)
+                )
+
+    def test_parsed_attributes(self):
+        """Testing that the attribute information parsed matches with the one captured
+        in the fandango json file.
+        """
+        actual_parsed_attrs = self.fandango_parser.get_device_attribute_metadata()
+        expected_attr_list = [
+            'Status', 'Timing_maximum', 'Timing_average',
+            'Timing_index', 'Timing_info', 'Timing_calls',
+            'State', 'StoredProcedureRelease', 'Timing_minimum'
+            ]
+        actual_parsed_attr_list = actual_parsed_attrs.keys()
+        self.assertGreater(
+            len(actual_parsed_attr_list), 0,
+            "There is no attribute information parsed"
+            )
+        self.assertEquals(
+            expected_attr_list.sort(), actual_parsed_attr_list.sort(),
+            "There are missing attributes"
+            )
+
+        # Using the 'Status' attribute expected results
+        self.assertIn(
+            'Status', actual_parsed_attrs.keys(),
+            "The attribute Status is not in the parsed attribute list"
+            )
+        actual_parsed_status_attr_info = actual_parsed_attrs['Status']
+        self.assertEquals(
+            len(actual_parsed_status_attr_info.keys()), len(EXPECTED_STATUS_ATTR_INFO.keys()),
+            "There are extra/missing properties in the parsed 'Status' attribute"
+            )
```

Comment:
You can assertEquals the sorted keys here instead of just checking the lengths. Then the failed assert output is more useful and it is also a better check. E.g.
+   self.assertEquals(
+            sorted(actual_parsed_status_attr_info.keys()), sorted(EXPECTED_STATUS_ATTR_INFO.keys()),
+            "There are extra/missing properties in the parsed 'Status' attribute"
+             )